tests/integration_tests/data_steward/validation/email_notification_test.py

```diff
@@ -32,7 +31,11 @@ def setUp(self):
             'submission_error': False
         }
 
-    def test_hpo_contact_list(self):
+    @mock.patch(
+        'validation.email_notification.query_sheet_linked_bq_table_app_engine')
+    def test_hpo_contact_list(self, query_rest_api):
+        query_rest_api.return_value = en.query_sheet_linked_bq_table_compute_engine(
```

Comment:
I recommended creating the drive-backed table within `EmailNotificationTest.setUp()`- but even overlooking the GDrive aspect and just loading a regular table (using `fake_dict` below) would help better encapsulate the test resources. If neither are feasible then indicate in a comment any special requirements of this test (e.g. account access to a particular spreadsheet,  existence of a table backed by the spreadsheet).

Reply:
I think `query_sheet_linked_bq_table_app_engine` and other sheet related functions should be in `utils/bq.py` instead of the email script. What do you think @mark-velez?

Reply:
I also found out that it cannot create a spreadsheet. A spreadsheet needs to be created separately and granted permissions to the service account. It can, however, link the spreadsheet to a table, and I have added that in the test.

Reply:
I think your inclination that reusable bq-related functions should be relocated to `utils.bq` is right. I recommend moving `post_query` and somehow making it explicit that the function is specialized to handle the scope issue (e.g. refactoring it to be `query_with_scopes(scopes: List[str])`, including a detailed doc string, etc.). I think we should avoid putting anything in `utils.bq` that regards specific tables as it is less likely to be more generally useful.

Reply:
I agree, this was my thought as well. I've made them all of them available for general usage.